media-gfx/gnofract4d/gnofract4d-4.3.ebuild

```diff
@@ -0,0 +1,53 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7,8} )
+DISTUTILS_SINGLE_IMPL=1
+
+inherit distutils-r1 eutils virtualx xdg
+
+DESCRIPTION="A program for drawing beautiful mathematically-based images known as fractals"
+HOMEPAGE="https://fract4d.github.io/gnofract4d/"
+SRC_URI="https://github.com/fract4d/${PN}/releases/download/v${PV}/${P}.tar.gz
+	https://github.com/fract4d/gnofract4d/raw/v4.3/testdata/test.ase -> ${PF}-test.ase"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+COMMON_DEPEND="
+	media-libs/libpng:0=
+	virtual/jpeg:0"
+RDEPEND="${COMMON_DEPEND}
+	$(python_gen_cond_dep '
+		dev-python/pycairo[${PYTHON_USEDEP}]
+		dev-python/pygobject:3[${PYTHON_USEDEP}]
+	')
+	x11-libs/gtk+:3[introspection]"
+BDEPEND="virtual/pkgconfig"
+DEPEND="${COMMON_DEPEND}"
+
+distutils_enable_tests pytest
+
+src_prepare() {
+	cp "${DISTDIR}/${PF}-test.ase" testdata/test.ase || die
+	distutils-r1_src_prepare
+	sed -i -e "s:share/doc/gnofract4d/:share/doc/${PF}/:" setup.py || die
+}
+
+python_test() {
+	virtx pytest fract4d fract4d_compiler fract4dgui
+}
+
+python_compile_all() {
+	if use test; then
+		ln -s "${BUILD_DIR}"/lib/fract4d/*.so fract4d/ || die
+	fi
+}
+
+pkg_postinst() {
+	xdg_pkg_postinst
```

Comment:
This is called automatically in xdg eclass

Reply:
Yes but pkg_postinst is there because of the optfeature all, which then means the call to xdg_pkg_postinst is needed I think.

Reply:
Optfeature is handled by the eutils eclass: https://devmanual.gentoo.org/eclass-reference/eutils.eclass/index.html